tests/resources/Harbor-Pages/Project-Members.robot

```diff
@@ -21,69 +21,254 @@ ${HARBOR_VERSION}  v1.1.1
+    Logout Harbor
+
+User A Create New Project And Push Image To It
+    [Arguments]  ${ip}  ${d}
+    Sign In Harbor  ${HARBOR_URL}  usera${d}  Test1@34
+    #create project
+    Create An New Project  project${d}
+    #verify can not change role
+    Go Into Project  project${d}
+    Switch To Member
+    User Can Not Change Role  usera${d}
+    Logout Harbor
+    #login console as usera and push
+    ${rc}=  Run And Return Rc  docker pull hello-world
+    ${rc}  ${output}=  Run And Return Rc And Output  docker login -u usera${d} -p Test1@34 ${ip}
+    ${rc}=  Run And Return Rc  docker tag hello-world ${ip}/project${d}/hello-world
+    ${rc}=  Run And Return Rc  docker push ${ip}/project${d}/hello-world
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}=  Run And Return Rc  docker logout ${ip}
+
+User B Not Project Member Cannot See And Operate The Project
+    [Arguments]  ${ip}  ${d}
+    #logout change userb and pull push
+    ${rc}  ${output}=  Run And Return Rc And Output  docker login -u userb${d} -p Test1@34 ${ip}
+    ${rc}=  Run And Return Rc  docker pull ${ip}/project${d}/hello-world
+    Should Not Be Equal As Integers  ${rc}  0  
+    ${rc}=  Run And Return Rc  docker tag hello-world ${ip}/project${d}/bbbbb
+    ${rc}=  Run And Return Rc  docker push ${ip}/project${d}/bbbbb
+    Should Not Be Equal As Integers  ${rc}  0
+    #login ui as b
+    Sign In Harbor  ${HARBOR_URL}  userb${d}  Test1@34
+    Project Should Not Display  project${d}
+    Logout Harbor
+ 
+User A Add User B As Guest Role On Project
+    [Arguments]  ${d}
+    #login as a
+    Sign In Harbor  ${HARBOR_URL}  usera${d}  Test1@34
+    Go Into Project  project${d}
+    Switch To Member
+    #click add member
+    Add Guest Member To Project  userb${d}
+    Logout Harbor
+
+User B As Guest Only Has Read Permission
+    [Arguments]  ${ip}  ${d}
+    #sign in as b
+    Sign In Harbor   ${HARBOR_URL}  userb${d}  Test1@34
+    #step 12
+    Project Should Display  project${d}
+    #step 13
+    Go Into Project  project${d}
+    Switch To Member
+    #page should contain element  xpath=//project-detail//clr-dg-cell//clr-dg-action-overflow[@hidden=""]
+    Non-admin View Member Account  2
+    #step 14
+    User Can Not Add Member
+    ${rc}  ${output}=  Run And Return Rc And Output  docker login -u userb${d} -p Test1@34 ${ip}
+    #step 15
+    ${rc}=  Run And Return Rc  docker pull ${ip}/project${d}/hello-world
+    #step 16
+    ${rc}=  Run And Return Rc  docker push ${ip}/project${d}/bbbbb
+    Should Not Be Equal As Integers  ${rc}  0
+    Logout Harbor
+
+User A Change Role of User B To Developer
+    [Arguments]  ${d}
+    Sign In Harbor  ${HARBOR_URL}  usera${d}  Test1@34
+    #change userb to developer
+    Change Project Member Role  project${d}  userb${d}  Developer
+    Logout Harbor
+
+User B As Developer Can Manage Image But Cannot Manage Member
```

Comment:
I do believe all the keywords are all reusable instead of a description of a piece of code.
Let's discuss.